templates/02-counter/src/main.js

```diff
@@ -19,9 +19,42 @@ function updateUI() {
     Array.from(document.querySelectorAll('.sign-in')).map(it => it.style = "display: block;");
   } else {
     Array.from(document.querySelectorAll('.after-sign-in')).map(it => it.style = "display: block;");
+    contract.getCounter().then(count => {
+      document.querySelector('#show').classList.remove('loader')
+      document.querySelector('#show').innerText = count == undefined ? "calculating..." : count
+      document.querySelector('#left').classList.toggle('eye')
+      Array.from(document.querySelectorAll('button')).map(it => it.disabled = false)
+    })
   }
 }
+// counter method
+var value = 1
 
+document.querySelector('#plus').addEventListener('click', ()=>{
+  Array.from(document.querySelectorAll('button')).map(it => it.disabled = true)
+  document.querySelector('#show').classList.add('loader')
+  contract.incrementCounter({value:value}).then(updateUI)
+})
+document.querySelector('#minus').addEventListener('click', ()=>{
+  Array.from(document.querySelectorAll('button')).map(it => it.disabled = true)
+  document.querySelector('#show').classList.add('loader')
+  contract.decrementCounter({value:value}).then(updateUI)
+})
+document.querySelector('#a').addEventListener('click', ()=>{
+  Array.from(document.querySelectorAll('button')).map(it => it.disabled = true)
+  document.querySelector('#show').classList.add('loader')
+  contract.resetCounter().then(updateUI)
+})
+document.querySelector('#c').addEventListener('click', ()=>{
+  document.querySelector('#left').classList.toggle('eye')
+})
+document.querySelector('#b').addEventListener('click', ()=>{
+  document.querySelector('#right').classList.toggle('eye')
+})
+document.querySelector('#d').addEventListener('click', ()=>{
+  document.querySelector('.dot').classList.toggle('on')
+  value = 10
```

Comment:
I'd expect this to toggle between 1 and 10 too (similar to how for is toggled on/off)